lib/membrane_webrtc_plugin/endpoint.ex

```diff
@@ -45,6 +45,24 @@ defmodule Membrane.WebRTC.Endpoint do
 
   @spec update_track_encoding(endpoint :: Endpoint.t(), track_id :: Track.id(), encoding :: atom) ::
           Endpoint.t()
-  def update_track_encoding(endpoint, track_id, value),
-    do: update_in(endpoint.inbound_tracks[track_id], &%Track{&1 | encoding: value})
+  def update_track_encoding(endpoint, track_id, value) do
+    update_in(endpoint.inbound_tracks[track_id], &%Track{&1 | encoding: value})
+  end
+
+  @spec update_tracks_ids(Membrane.WebRTC.Endpoint.t(), map) :: Membrane.WebRTC.Endpoint.t()
+  def update_tracks_ids(endpoint, mid_to_id) do
+    inbound_tracks =
+      endpoint.inbound_tracks
+      |> Map.values()
+      |> Enum.map(fn %Track{mid: track_mid} = track ->
+        with %{^track_mid => track_id} <- mid_to_id do
+          %Track{track | id: track_id}
+        else
+          _mid_to_id -> track
+        end
+      end)
+      |> Map.new(&{&1.id, &1})
```

Comment:
If you decided to change `mid_to_id` to a tuple `{mid, track_id}` this could look like this:
```suggestion
      |> Enum.map(fn 
        %Track{mid: ^mid} = track ->
            %Track{track | id: track_id}
        track ->
            track
      end)
      |> Map.new(&{&1.id, &1})
```

Reply:
Maybe sth like this? 
```elixir
  @spec update_tracks_ids(t(), map) :: t()
  def update_tracks_ids(endpoint, mid_to_id) do
    inbound_tracks =
      Map.new(endpoint.inbound_tracks, fn {track_id, track} ->
        new_track_id = Map.get(mid_to_id, track.mid)
        {new_track_id, %Track{track | id: track_id}}
      end)

    %__MODULE__{endpoint | inbound_tracks: inbound_tracks}
  end
  ```

Reply:
But this way we always have to update all inbound tracks so it might not be the best idea. I would leave it as it is but replace Map.values and Enum.map with single Map.new in this way
```elixir
  @spec update_tracks_ids(t(), map) :: t()
  def update_tracks_ids(endpoint, mid_to_id) do
    inbound_tracks =
      endpoint.inbound_tracks
      |> Map.new(fn {track_id, %Track{mid: track_mid}} = track ->
        with %{^track_mid => new_track_id} <- mid_to_id do
          {new_track_id, %Track{track | id: new_track_id}}
        else
          _mid_to_id -> {track_id, track}
        end
      end)

    %__MODULE__{endpoint | inbound_tracks: inbound_tracks}
  end
```